camera/tests/test_net_requests.py

```diff
@@ -0,0 +1,124 @@
+import math
+import numpy as np
+from src.net_requests import *
+import responses
+import unittest
+import time
+import base64
+
+class TestNetRequests(unittest.TestCase):
+    def setUp(self) -> None:
+        return super().setUp()
+    
+    @responses.activate
+    def test_heartbeat_timer(self):
+        id = "111"
+        call_count = 3
+        heartbeat_delay = 0.1
+
+        resp = responses.put(
+            NetConfig.get_heartbeat_url(id),
+            headers={
+                "Content-Type": "application/json",
+                "Authorization": "aaaa"
+            }
+        )
+
+        heartbeat = Heartbeat(heartbeat_delay=heartbeat_delay)
+        sleep(heartbeat_delay * call_count)
+        heartbeat.stop()
+        self.assertTrue(resp.call_count == call_count)
+    
+    @responses.activate
+    def test_heartbeat_request(self):
+        id = "111"
+
+        cur_time = time.time()
+        responses.put(
+            NetConfig.get_heartbeat_url(id),
+            headers={
```

Comment:
nit: I don't think the `heartbeat`or `notifications` endpoint will return any headers or data after a successful `PUT` request, so returning mock headers and mock JSON isn't necessary in these tests. I didn't really say that or right it down anywhere though, so I'm fine with how it is now if you just want to leave it as-is

Reply:
From what I understand, it's matching that header with any requests sent. This isn't the response given back afaik.